aten/src/ATen/native/xnnpack/Clamp.cpp

```diff
@@ -0,0 +1,194 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Allocator.h>
+#include <ATen/native/xnnpack/Common.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace clamp {
+
+struct Context final {
+  Operator clamp_op;
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32
+//  - relu
+//  - relu6
+//  - leaky relu
+//  - hard tanh
+//  - or any other operator that can be expressed as a clamp
+
+bool available(
+    const int64_t input_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // XNNPACK
+  return xnnpack::internal::available() &&
+         // Input == Output
+         (input_channels == output_channels) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
+}
+
+Context create(
+    const int64_t input_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+  TORCH_CHECK(
+      available(input_channels, output_channels, output_min, output_max),
+      "xnnpack::clamp not available!");
+
+  xnn_operator_t clamp_op{};
+
+  const xnn_status create_status = xnn_create_clamp_nc_f32(
+      input_channels,         // channels,
+      input_channels,         // input_pixel_stride
```

Comment:
you cannot put random value in here just to make xnn_create_clamp_nc_f32 pass, you should put actual corresponding stride of the input tensor.

Reply:
The stride is a multiple of channels not bytes, and if I am not mistaken denotes the distance between two adjacent batches.  Do we have tensors with batches not adjacent in memory?  I assume that might be the difference between contiguous and non-contiguous tensors?  How can I query the stride in that case?

Reply:
@Maratyszcza Is my understanding of XNNPACK's use of the stride terminology correct?

Reply:
tensor.strides() gives the strides of the tensor. For contiguous tensors (NHWC - contiguous tensors in your case) input_pixel_stride will be equal to the number of channels, but in general it's not guaranteed.

Reply:
`input_pixel_stride`/`output_pixel_stride` is stride, in number of elements, between neighboring pixels along the `channel` dimesion in NHWC.

Reply:
XNNPACK requires that NHW dimensions are contiguous (for Clamp they are all merged into `batch_size` argument), but C stride can be larger (but not smaller) than the number of channels.

Reply:
Removing Clamp.